diff --git a/src/freedreno/common/freedreno_dev_info.h b/src/freedreno/common/freedreno_dev_info.h
index a1299a0dc09..8d07c7ca566 100644
--- a/src/freedreno/common/freedreno_dev_info.h
+++ b/src/freedreno/common/freedreno_dev_info.h
@@ -468,12 +468,12 @@ struct fd_dev_info {
        * expected:
        */
       bool has_salu_int_narrowing_quirk;
-
       /* Whether the device supports the image processing opcode */
       bool has_image_processing;
-
       /* The amount of valid draw state IDs. */
       uint32_t max_draw_states;
+      /* If GMEM needs to be disabled for this GPU */
+      bool disable_gmem;
    } props;
 };
 
diff --git a/src/freedreno/common/freedreno_devices.py b/src/freedreno/common/freedreno_devices.py
index 80bc917221c..061bbbb5d99 100644
--- a/src/freedreno/common/freedreno_devices.py
+++ b/src/freedreno/common/freedreno_devices.py
@@ -967,6 +967,53 @@ a740_raw_magic_regs = [
         [A6XXRegs.REG_A6XX_UCHE_UNKNOWN_0E12, 0],
     ]
 
+
+# Adreno 710/720 are not supported by the upstream, but some hacks float on the internet adding their support.
+# These hacks simply reuse A730 entry with different ids and looks like it works in some extent
+# Let's do the same in our patchset
+add_gpus([
+        GPUId(chip_id=0x07010000, name="FD710"), # KGSL, no speedbin data
+        GPUId(chip_id=0xffff07010000, name="FD710"), # Default no-speedbin fallback
+    ], A6xxGPUInfo(
+        CHIP.A7XX,
+        [a7xx_base, a7xx_gen1],
+        num_ccu = 4,
+        tile_align_w = 64,
+        tile_align_h = 32,
+        tile_max_w = 1024,
+        tile_max_h = 1024,
+        num_vsc_pipes = 32,
+        cs_shared_mem_size = 32 * 1024,
+        wave_granularity = 2,
+        fibers_per_sp = 128 * 2 * 16,
+        highest_bank_bit = 16,
+        magic_regs = a730_magic_regs,
+        raw_magic_regs = a730_raw_magic_regs,
+    ))
+
+# Adreno 720
+add_gpus([
+        GPUId(chip_id=0x43020000, name="FD720"), # KGSL, no speedbin data
+        GPUId(chip_id=0xffff43020000, name="FD720"), # Default no-speedbin fallback
+    ], A6xxGPUInfo(
+        CHIP.A7XX,
+        [a7xx_base, a7xx_gen1],
+        num_ccu = 4,
+        tile_align_w = 64,
+        tile_align_h = 32,
+        tile_max_w = 1024,
+        tile_max_h = 1024,
+        num_vsc_pipes = 32,
+        cs_shared_mem_size = 32 * 1024,
+        wave_granularity = 2,
+        fibers_per_sp = 128 * 2 * 16,
+        highest_bank_bit = 16,
+        magic_regs = a730_magic_regs,
+        raw_magic_regs = a730_raw_magic_regs,
+    ))
+
+
+
 add_gpus([
         # These are named as Adreno730v3 or Adreno725v1.
         GPUId(chip_id=0x07030002, name="FD725"),
@@ -1376,6 +1423,32 @@ add_gpus([
         raw_magic_regs = a8xx_base_raw_magic_regs,
     ))
 
+# gen8_6_0
+add_gpus([
+        GPUId(chip_id=0x44030000, name="Adreno (TM) 825"),
+    ], A6xxGPUInfo(
+        CHIP.A8XX,
+        [a7xx_base, a7xx_gen3, a8xx_base, a8xx_gen1, GPUProps(
+            # This is probably not an optimal config for gmem/sysmem, but it was working before and I don't have any a825 device to test (neither I have any trace info)
+            sysmem_ccu_color_cache_fraction = CCUColorCacheFraction.FULL.value,
+            sysmem_per_ccu_color_cache_size = 128 * 1024,
+            sysmem_ccu_depth_cache_fraction = CCUColorCacheFraction.THREE_QUARTER.value,
+            sysmem_per_ccu_depth_cache_size = 96 * 1024,
+        )],
+        num_ccu = 4,
+        num_slices = 2,
+        tile_align_w = 96,
+        tile_align_h = 32,
+        tile_max_w = 16416,
+        tile_max_h = 16384,
+        num_vsc_pipes = 32,
+        cs_shared_mem_size = 32 * 1024,
+        wave_granularity = 2,
+        fibers_per_sp = 128 * 2 * 16,
+        magic_regs = dict(),
+        raw_magic_regs = a8xx_base_raw_magic_regs,
+    ))
+
 add_gpus([
         GPUId(chip_id=0x44030a20, name="Adreno (TM) 829"), # KGSL
     ], A6xxGPUInfo(
diff --git a/src/freedreno/drm-shim/freedreno_noop.c b/src/freedreno/drm-shim/freedreno_noop.c
index 5541c6cd90e..73fd3d6ff4f 100644
--- a/src/freedreno/drm-shim/freedreno_noop.c
+++ b/src/freedreno/drm-shim/freedreno_noop.c
@@ -286,6 +286,21 @@ static const struct msm_device_info device_infos[] = {
       .chip_id = 0x44050000,
       .gmem_size = 12 * 1024 * 1024,
    },
+   {
+      .gpu_id = 825,
+      .chip_id = 0x44030000,
+      .gmem_size = 2 * 1024 * 1024,
+   },
+   {
+      .gpu_id = 829,
+      .chip_id = 0x44030A20,
+      .gmem_size = 2 * 1024 * 1024,
+   },
+   {
+      .gpu_id = 810,
+      .chip_id = 0x44010000,
+      .gmem_size = 576 * 1024,
+   }
 };
 
 static void
diff --git a/src/freedreno/perfcntrs/fd8_perfcntr.c b/src/freedreno/perfcntrs/fd8_perfcntr.c
new file mode 100644
index 00000000000..51955e07996
--- /dev/null
+++ b/src/freedreno/perfcntrs/fd8_perfcntr.c
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2026 Igalia S.L.
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "util/u_math.h"
+
+#include "fd6_hw.h"
+
+#include "freedreno_perfcntr.h"
+
+#define REG8(_x) REG_A8XX_##_x
+
+#define FD8_COUNTER(_sel, _offset) COUNTER_BASE(_sel, _offset+0, _offset+1)
+#define FD8_COUNTABLE(_sel) COUNTABLE_BASE(#_sel, A7XX_##_sel, UINT64, AVERAGE)
+
+/* FIXME: this is a limited listing of perfcntr counter groups on a8xx,
+ * only specifying the CP group counter registers and countable values
+ * which are used by the new autotuner implementation. The perfcntr data
+ * will also be autogenerated in the near future, so adding complete a8xx
+ * support should be done under that effort.
+ *
+ * CP counter registers have changed on a8xx, but we can reuse the a7xx
+ * countable list since we don't have the a8xx-specific list yet, and the
+ * autotuner is only interested in countables that should be common between
+ * a7xx and a8xx.
+ */
+
+static const struct fd_perfcntr_counter cp_counters[] = {
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(0)), REG8(RBBM_PERFCTR_CP(0))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(1)), REG8(RBBM_PERFCTR_CP(1))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(2)), REG8(RBBM_PERFCTR_CP(2))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(3)), REG8(RBBM_PERFCTR_CP(3))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(4)), REG8(RBBM_PERFCTR_CP(4))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(5)), REG8(RBBM_PERFCTR_CP(5))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(6)), REG8(RBBM_PERFCTR_CP(6))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(7)), REG8(RBBM_PERFCTR_CP(7))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(8)), REG8(RBBM_PERFCTR_CP(8))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(9)), REG8(RBBM_PERFCTR_CP(9))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(10)), REG8(RBBM_PERFCTR_CP(10))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(11)), REG8(RBBM_PERFCTR_CP(11))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(12)), REG8(RBBM_PERFCTR_CP(12))),
+      FD8_COUNTER(REG8(CP_PERFCTR_CP_SEL(13)), REG8(RBBM_PERFCTR_CP(13))),
+};
+
+static const struct fd_perfcntr_countable cp_countables[] = {
+      FD8_COUNTABLE(PERF_CP_ALWAYS_COUNT),
+      FD8_COUNTABLE(PERF_CP_BUSY_GFX_CORE_IDLE),
+      FD8_COUNTABLE(PERF_CP_BUSY_CYCLES),
+      FD8_COUNTABLE(PERF_CP_NUM_PREEMPTIONS),
+      FD8_COUNTABLE(PERF_CP_PREEMPTION_REACTION_DELAY),
+      FD8_COUNTABLE(PERF_CP_PREEMPTION_SWITCH_OUT_TIME),
+      FD8_COUNTABLE(PERF_CP_PREEMPTION_SWITCH_IN_TIME),
+      FD8_COUNTABLE(PERF_CP_DEAD_DRAWS_IN_BIN_RENDER),
+      FD8_COUNTABLE(PERF_CP_PREDICATED_DRAWS_KILLED),
+      FD8_COUNTABLE(PERF_CP_MODE_SWITCH),
+      FD8_COUNTABLE(PERF_CP_ZPASS_DONE),
+      FD8_COUNTABLE(PERF_CP_CONTEXT_DONE),
+      FD8_COUNTABLE(PERF_CP_CACHE_FLUSH),
+      FD8_COUNTABLE(PERF_CP_LONG_PREEMPTIONS),
+      FD8_COUNTABLE(PERF_CP_SQE_I_CACHE_STARVE),
+      FD8_COUNTABLE(PERF_CP_SQE_IDLE),
+      FD8_COUNTABLE(PERF_CP_SQE_PM4_STARVE_RB_IB),
+      FD8_COUNTABLE(PERF_CP_SQE_PM4_STARVE_SDS),
+      FD8_COUNTABLE(PERF_CP_SQE_MRB_STARVE),
+      FD8_COUNTABLE(PERF_CP_SQE_RRB_STARVE),
+      FD8_COUNTABLE(PERF_CP_SQE_VSD_STARVE),
+      FD8_COUNTABLE(PERF_CP_VSD_DECODE_STARVE),
+      FD8_COUNTABLE(PERF_CP_SQE_PIPE_OUT_STALL),
+      FD8_COUNTABLE(PERF_CP_SQE_SYNC_STALL),
+      FD8_COUNTABLE(PERF_CP_SQE_PM4_WFI_STALL),
+      FD8_COUNTABLE(PERF_CP_SQE_SYS_WFI_STALL),
+      FD8_COUNTABLE(PERF_CP_SQE_T4_EXEC),
+      FD8_COUNTABLE(PERF_CP_SQE_LOAD_STATE_EXEC),
+      FD8_COUNTABLE(PERF_CP_SQE_SAVE_SDS_STATE),
+      FD8_COUNTABLE(PERF_CP_SQE_DRAW_EXEC),
+      FD8_COUNTABLE(PERF_CP_SQE_CTXT_REG_BUNCH_EXEC),
+      FD8_COUNTABLE(PERF_CP_SQE_EXEC_PROFILED),
+      FD8_COUNTABLE(PERF_CP_MEMORY_POOL_EMPTY),
+      FD8_COUNTABLE(PERF_CP_MEMORY_POOL_SYNC_STALL),
+      FD8_COUNTABLE(PERF_CP_MEMORY_POOL_ABOVE_THRESH),
+      FD8_COUNTABLE(PERF_CP_AHB_WR_STALL_PRE_DRAWS),
+      FD8_COUNTABLE(PERF_CP_AHB_STALL_SQE_GMU),
+      FD8_COUNTABLE(PERF_CP_AHB_STALL_SQE_WR_OTHER),
+      FD8_COUNTABLE(PERF_CP_AHB_STALL_SQE_RD_OTHER),
+      FD8_COUNTABLE(PERF_CP_CLUSTER0_EMPTY),
+      FD8_COUNTABLE(PERF_CP_CLUSTER1_EMPTY),
+      FD8_COUNTABLE(PERF_CP_CLUSTER2_EMPTY),
+      FD8_COUNTABLE(PERF_CP_CLUSTER3_EMPTY),
+      FD8_COUNTABLE(PERF_CP_CLUSTER4_EMPTY),
+      FD8_COUNTABLE(PERF_CP_CLUSTER5_EMPTY),
+      FD8_COUNTABLE(PERF_CP_PM4_DATA),
+      FD8_COUNTABLE(PERF_CP_PM4_HEADERS),
+      FD8_COUNTABLE(PERF_CP_VBIF_READ_BEATS),
+      FD8_COUNTABLE(PERF_CP_VBIF_WRITE_BEATS),
+      FD8_COUNTABLE(PERF_CP_SQE_INSTR_COUNTER),
+      FD8_COUNTABLE(PERF_CP_RESERVED_50),
+      FD8_COUNTABLE(PERF_CP_RESERVED_51),
+      FD8_COUNTABLE(PERF_CP_RESERVED_52),
+      FD8_COUNTABLE(PERF_CP_RESERVED_53),
+      FD8_COUNTABLE(PERF_CP_RESERVED_54),
+      FD8_COUNTABLE(PERF_CP_RESERVED_55),
+      FD8_COUNTABLE(PERF_CP_RESERVED_56),
+      FD8_COUNTABLE(PERF_CP_RESERVED_57),
+      FD8_COUNTABLE(PERF_CP_RESERVED_58),
+      FD8_COUNTABLE(PERF_CP_RESERVED_59),
+      FD8_COUNTABLE(PERF_CP_CLUSTER0_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER1_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER2_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER3_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER4_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER5_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER6_FULL),
+      FD8_COUNTABLE(PERF_CP_CLUSTER6_EMPTY),
+      FD8_COUNTABLE(PERF_CP_ICACHE_MISSES),
+      FD8_COUNTABLE(PERF_CP_ICACHE_HITS),
+      FD8_COUNTABLE(PERF_CP_ICACHE_STALL),
+      FD8_COUNTABLE(PERF_CP_DCACHE_MISSES),
+      FD8_COUNTABLE(PERF_CP_DCACHE_HITS),
+      FD8_COUNTABLE(PERF_CP_DCACHE_STALLS),
+      FD8_COUNTABLE(PERF_CP_AQE_SQE_STALL),
+      FD8_COUNTABLE(PERF_CP_SQE_AQE_STARVE),
+      FD8_COUNTABLE(PERF_CP_PREEMPT_LATENCY),
+      FD8_COUNTABLE(PERF_CP_SQE_MD8_STALL_CYCLES),
+      FD8_COUNTABLE(PERF_CP_SQE_MESH_EXEC_CYCLES),
+      FD8_COUNTABLE(PERF_CP_AQE_NUM_AS_CHUNKS),
+      FD8_COUNTABLE(PERF_CP_AQE_NUM_MS_CHUNKS),
+};
+
+const struct fd_perfcntr_group a8xx_perfcntr_groups[] = {
+      GROUP("CP", cp_counters, cp_countables),
+};
+
+const unsigned a8xx_num_perfcntr_groups = ARRAY_SIZE(a8xx_perfcntr_groups);
diff --git a/src/freedreno/perfcntrs/freedreno_perfcntr.c b/src/freedreno/perfcntrs/freedreno_perfcntr.c
index 7c615e930f4..b8db821be69 100644
--- a/src/freedreno/perfcntrs/freedreno_perfcntr.c
+++ b/src/freedreno/perfcntrs/freedreno_perfcntr.c
@@ -22,6 +22,9 @@ extern const unsigned a6xx_num_perfcntr_groups;
 extern const struct fd_perfcntr_group a7xx_perfcntr_groups[];
 extern const unsigned a7xx_num_perfcntr_groups;
 
+extern const struct fd_perfcntr_group a8xx_perfcntr_groups[];
+extern const unsigned a8xx_num_perfcntr_groups;
+
 const struct fd_perfcntr_group *
 fd_perfcntrs(const struct fd_dev_id *id, unsigned *count)
 {
@@ -38,6 +41,9 @@ fd_perfcntrs(const struct fd_dev_id *id, unsigned *count)
    case 7:
       *count = a7xx_num_perfcntr_groups;
       return a7xx_perfcntr_groups;
+   case 8:
+      *count = a8xx_num_perfcntr_groups;
+      return a8xx_perfcntr_groups;
    default:
       *count = 0;
       return NULL;
diff --git a/src/freedreno/perfcntrs/meson.build b/src/freedreno/perfcntrs/meson.build
index bef5ec72520..5b0dbf8a0d2 100644
--- a/src/freedreno/perfcntrs/meson.build
+++ b/src/freedreno/perfcntrs/meson.build
@@ -6,6 +6,7 @@ libfreedreno_perfcntrs_files = files(
   'fd5_perfcntr.c',
   'fd6_perfcntr.c',
   'fd7_perfcntr.c',
+  'fd8_perfcntr.c',
   'freedreno_dt.c',
   'freedreno_dt.h',
   'freedreno_perfcntr.c',
diff --git a/src/freedreno/vulkan/tu_cmd_buffer.cc b/src/freedreno/vulkan/tu_cmd_buffer.cc
index ddcc34382a4..965c767c47e 100644
--- a/src/freedreno/vulkan/tu_cmd_buffer.cc
+++ b/src/freedreno/vulkan/tu_cmd_buffer.cc
@@ -1323,6 +1323,12 @@ use_sysmem_rendering(struct tu_cmd_buffer *cmd,
       return true;
    }
 
+   bool no_gmem = cmd->device->physical_device->dev_info.props.disable_gmem;
+   if (no_gmem) {
+       cmd->state.rp.gmem_disable_reason = "Unsupported GPU";
+       return true;
+    }
+
    /* can't fit attachments into gmem */
    if (!cmd->state.tiling->possible) {
       cmd->state.rp.gmem_disable_reason = "Can't fit attachments into gmem";
@@ -8733,7 +8739,7 @@ tu_CmdDrawIndirect(VkCommandBuffer commandBuffer,
 
    tu6_emit_empty_vs_params<CHIP>(cmd);
 
-   if (cmd->device->physical_device->info->props.indirect_draw_wfm_quirk)
+   if (cmd->device->physical_device->info->props.indirect_draw_wfm_quirk || CHIP >= A8XX)
       draw_wfm(cmd);
 
    tu6_draw_common<CHIP>(cmd, cs, false, 0);
@@ -8764,7 +8770,7 @@ tu_CmdDrawIndexedIndirect(VkCommandBuffer commandBuffer,
 
    tu6_emit_empty_vs_params<CHIP>(cmd);
 
-   if (cmd->device->physical_device->info->props.indirect_draw_wfm_quirk)
+   if (cmd->device->physical_device->info->props.indirect_draw_wfm_quirk || CHIP >= A8XX)
       draw_wfm(cmd);
 
    tu6_draw_common<CHIP>(cmd, cs, true, 0);
diff --git a/src/freedreno/vulkan/tu_device.cc b/src/freedreno/vulkan/tu_device.cc
index 7ac080b6dca..8b57ace0fc6 100644
--- a/src/freedreno/vulkan/tu_device.cc
+++ b/src/freedreno/vulkan/tu_device.cc
@@ -42,6 +42,9 @@
 #include "tu_tracepoints.h"
 #include "tu_wsi.h"
 
+#include "git_sha1.h"
+#include "tu_version.h"
+
 #if DETECT_OS_ANDROID
 #include <vndk/hardware_buffer.h>
 #endif
@@ -951,6 +954,12 @@ tu_get_physical_device_properties_1_2(struct tu_physical_device *pdevice,
       };
    }
 
+   if (TU_DEBUG(DECK_EMU)) {
+      p->driverID = VK_DRIVER_ID_MESA_RADV;
+      memset(p->driverName, 0, sizeof(p->driverName));
+      snprintf(p->driverName, VK_MAX_DRIVER_NAME_SIZE, "radv");
+   }
+
    p->denormBehaviorIndependence =
       VK_SHADER_FLOAT_CONTROLS_INDEPENDENCE_ALL;
    p->roundingModeIndependence =
@@ -1233,17 +1242,35 @@ tu_get_properties(struct tu_physical_device *pdevice,
    props->optimalBufferCopyRowPitchAlignment = 128;
    props->nonCoherentAtomSize = 64;
 
-   props->apiVersion =
+   /* HACK: Expose Vulkan 1.3 on devices without multiview.
+    * Current Minecraft renderer checks for VK1.2 presence and refuses to
+    * start on VK1.0 exposed here in the case if the device has no multiview
+    * This makes it boot on these devices. This is not conformant, but I don't
+    * care. Sigh.
+    */
+
+   /*props->apiVersion =
       tu_has_multiview(pdevice) ?
          ((pdevice->info->chip >= 7) ? TU_API_VERSION :
             VK_MAKE_VERSION(1, 3, VK_HEADER_VERSION))
          : VK_MAKE_VERSION(1, 0, VK_HEADER_VERSION);
+   */
+
+   props->apiVersion = pdevice->info->chip >= 7 ?
+      TU_API_VERSION : // expose current api on gen7 devices anyway
+      VK_MAKE_VERSION(1, 3, VK_HEADER_VERSION);
+
    props->driverVersion = vk_get_driver_version();
    props->vendorID = pdevice->instance->force_vk_vendor != 0 ?
                      pdevice->instance->force_vk_vendor : 0x5143;
    props->deviceID = pdevice->dev_id.chip_id;
    props->deviceType = VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU;
 
+   if (TU_DEBUG(DECK_EMU)) {
+      props->vendorID = 0x1002;
+      props->deviceID = 0x163F;
+   }
+
    /* Vulkan 1.4 */
    props->dynamicRenderingLocalReadDepthStencilAttachments = true;
    props->dynamicRenderingLocalReadMultisampledAttachments = true;
@@ -1255,9 +1282,16 @@ tu_get_properties(struct tu_physical_device *pdevice,
    props->sparseResidencyAlignedMipSize = false;
    props->sparseResidencyNonResidentStrict = true;
 
-   strcpy(props->deviceName, pdevice->name);
+   char devname[128];
+   strcpy(devname, pdevice->name);
+   strcat(devname, MESA_GIT_SHA1 "/" TUGEN8_DRV_VERSION);
+   strcpy(props->deviceName, devname);
    memcpy(props->pipelineCacheUUID, pdevice->cache_uuid, VK_UUID_SIZE);
 
+   if (TU_DEBUG(DECK_EMU)) {
+      strcpy(props->deviceName, "AMD Custom GPU 0405 (RADV VANGOGH)");
+   }
+
    tu_get_physical_device_properties_1_1(pdevice, props);
    tu_get_physical_device_properties_1_2(pdevice, props);
    tu_get_physical_device_properties_1_3(pdevice, props);
@@ -2776,8 +2810,6 @@ tu_CreateDevice(VkPhysicalDevice physicalDevice,
          &dispatch_table, &tu_device_entrypoints_a7xx, false);
       break;
    case 8:
-      /* gen8 TODO: */
-      tu_env.debug |= TU_DEBUG_FLUSHALL;  /* dEQP-VK.draw.\*from_compute\* */
       vk_device_dispatch_table_from_entrypoints(
          &dispatch_table, &tu_device_entrypoints_a8xx, false);
    }
diff --git a/src/freedreno/vulkan/tu_util.cc b/src/freedreno/vulkan/tu_util.cc
index f0c19c3465d..40dbfe063eb 100644
--- a/src/freedreno/vulkan/tu_util.cc
+++ b/src/freedreno/vulkan/tu_util.cc
@@ -56,6 +56,7 @@ static const struct debug_control tu_debug_options[] = {
    { "nofdm", TU_DEBUG_NOFDM },
    { "nocb", TU_DEBUG_NO_CONCURRENT_BINNING },
    { "forcecb", TU_DEBUG_FORCE_CONCURRENT_BINNING },
+   { "deck_emu", TU_DEBUG_DECK_EMU },
    { NULL, 0 }
 };
 
diff --git a/src/freedreno/vulkan/tu_util.h b/src/freedreno/vulkan/tu_util.h
index cfa18f34978..44a2d1ff2c5 100644
--- a/src/freedreno/vulkan/tu_util.h
+++ b/src/freedreno/vulkan/tu_util.h
@@ -74,6 +74,7 @@ enum tu_debug_flags : uint64_t
    TU_DEBUG_NOFDM                    = BITFIELD64_BIT(34),
    TU_DEBUG_NO_CONCURRENT_BINNING    = BITFIELD64_BIT(35),
    TU_DEBUG_FORCE_CONCURRENT_BINNING = BITFIELD64_BIT(36),
+   TU_DEBUG_DECK_EMU                 = BITFIELD64_BIT(37),
 };
 
 struct tu_env {
diff --git a/src/freedreno/vulkan/tu_version.h b/src/freedreno/vulkan/tu_version.h
new file mode 100644
index 00000000000..e7c9e688c49
--- /dev/null
+++ b/src/freedreno/vulkan/tu_version.h
@@ -0,0 +1 @@
+#define TUGEN8_DRV_VERSION "unknown"
