diff --git a/src/freedreno/common/freedreno_dev_info.h b/src/freedreno/common/freedreno_dev_info.h
index 8d07c7ca566..c9581baf394 100644
--- a/src/freedreno/common/freedreno_dev_info.h
+++ b/src/freedreno/common/freedreno_dev_info.h
@@ -474,6 +474,9 @@ struct fd_dev_info {
       uint32_t max_draw_states;
       /* If GMEM needs to be disabled for this GPU */
       bool disable_gmem;
+
+      /* GMEM size in bytes */
+      uint32_t gmem_size;
    } props;
 };
 
diff --git a/src/freedreno/common/freedreno_devices.py b/src/freedreno/common/freedreno_devices.py
index 061bbbb5d99..577aa9fc864 100644
--- a/src/freedreno/common/freedreno_devices.py
+++ b/src/freedreno/common/freedreno_devices.py
@@ -1297,11 +1297,50 @@ add_gpus([
     ))
 
 a8xx_base = GPUProps(
+        prim_alloc_threshold = 0x7,
+        vs_max_inputs_count = 32,
+        max_sets = 8,
+        instr_cache_size = 127,
+        tess_use_shared = True,
+        has_per_view_viewport = True,
+        has_gmem_vpc_attr_buf = True,
+        has_lpac = True,
+        has_getfiberid = True,
+        has_movs = True,
+        has_dp4acc = True,
+        has_compliant_dp4acc = True,
+        has_scalar_alu = True,
+        has_scalar_predicates = True,
+        has_sad = True,
+        has_ssbo_imm_offsets = True,
+        has_64b_ssbo_atomics = True,
+        ubwc_unorm_snorm_int_compatible = True,
+        ubwc_all_formats_compatible = True,
+        supports_uav_ubwc = True,
+        has_ubwc_linear_mipmap_fallback = True,
+        supports_linear_mipmap_threshold_in_blocks = True,
+        has_early_preamble = True,
+        load_shader_consts_via_preamble = True,
+        load_inline_uniforms_via_preamble_ldgk = True,
+        r8g8_faulty_fast_clear_quirk = True,
+        gs_vpc_adjacency_quirk = True,
+        has_gmem_fast_clear = True,
+        has_hw_multiview = True,
+        has_fs_tex_prefetch = True,
+        has_sampler_minmax = True,
+        has_astc_hdr = True,
+        has_z24uint_s8uint = True,
+        has_sample_locations = True,
+        has_attachment_shading_rate = True,
+        has_primitive_shading_rate = True,
+        has_generic_clear = True,
+        has_alias_rt = True,
         has_dp2acc = False,
         reg_size_vec4 = 96,
         has_rt_workaround = False,
-        supports_double_threadsize = False,
+        supports_double_threadsize = True,
         has_dual_wave_dispatch = True,
+        has_salu_int_narrowing_quirk = False,
     )
 
 # For a8xx, the chicken bit and most other non-ctx reg
@@ -1381,18 +1420,41 @@ add_gpus([
        GPUId(chip_id=0xffff44010000, name="Adreno (TM) 810"),
     ], A6xxGPUInfo(
         CHIP.A8XX,
-        [a7xx_base, a7xx_gen3, a8xx_base, a8xx_gen1, GPUProps(
+        [a7xx_base, a7xx_gen3, a8xx_base, a8xx_gen2, GPUProps(
+            # Sysmem буфферы
+            sysmem_vpc_attr_buf_size = 131072, 
+            sysmem_vpc_pos_buf_size = 65536,
+            sysmem_vpc_bv_pos_buf_size = 32768,
+# глубины цветов sysmem
+            sysmem_ccu_color_cache_fraction = CCUColorCacheFraction.FULL.value,
+            sysmem_per_ccu_color_cache_size = 64 * 1024,
+            sysmem_ccu_depth_cache_fraction = CCUColorCacheFraction.THREE_QUARTER.value,
+            sysmem_per_ccu_depth_cache_size = 64 * 1024,
+#Gmem глубина и цвет 
+            gmem_ccu_color_cache_fraction = CCUColorCacheFraction.EIGHTH.value,
+            gmem_per_ccu_color_cache_size = 32 * 1024,
+            gmem_ccu_depth_cache_fraction = CCUColorCacheFraction.FULL.value,
+            gmem_per_ccu_depth_cache_size = 48 * 1024,
+            
+            #Gmem буфферы
             gmem_vpc_attr_buf_size = 16384,
             gmem_vpc_pos_buf_size = 12288,
             gmem_vpc_bv_pos_buf_size = 20480,
-            # This is possibly also needed for a830 (and all of a8xx),
-            # move to a8xx_base if confirmed needed for a830.
+
+            reg_size_vec4 = 96, # Для 810 лучше подходить 96, хоть оно и относится ко 2 поколению
+            disable_gmem = False,
+            gmem_size = 576 * 1024, # Слишком мало...
+            has_ray_intersection = False,
+            has_sw_fuse = False,
+            has_coherent_ubwc_flag_caches = True,
             has_fs_tex_prefetch = False,
+            has_salu_int_narrowing_quirk = True,
+            shading_rate_matches_vk = True,
         )],
         num_ccu = 1,
         num_slices = 1,
-        tile_align_w = 32,
-        tile_align_h = 16,
+        tile_align_w = 64,
+        tile_align_h = 32,
         tile_max_w = 16384,
         tile_max_h = 16384,
         num_vsc_pipes = 32,
@@ -1429,15 +1491,36 @@ add_gpus([
     ], A6xxGPUInfo(
         CHIP.A8XX,
         [a7xx_base, a7xx_gen3, a8xx_base, a8xx_gen1, GPUProps(
+            #Gmem глубина и цвет
+            gmem_ccu_color_cache_fraction = CCUColorCacheFraction.HALF.value,
+            gmem_per_ccu_color_cache_size = 128 * 1024,
+            gmem_ccu_depth_cache_fraction = CCUColorCacheFraction.HALF.value,
+            gmem_per_ccu_depth_cache_size = 128 * 1024,
+            #sysmem глубина и цвета 
             # This is probably not an optimal config for gmem/sysmem, but it was working before and I don't have any a825 device to test (neither I have any trace info)
+
             sysmem_ccu_color_cache_fraction = CCUColorCacheFraction.FULL.value,
             sysmem_per_ccu_color_cache_size = 128 * 1024,
             sysmem_ccu_depth_cache_fraction = CCUColorCacheFraction.THREE_QUARTER.value,
             sysmem_per_ccu_depth_cache_size = 96 * 1024,
+            #Sysmem кэши 
+             sysmem_vpc_attr_buf_size  = 131072,
+            
+            # Gmem кэши
+            gmem_vpc_attr_buf_size = 65536,              # 64 KB 
+            gmem_vpc_pos_buf_size = 32768,                # 32 KB
+            gmem_vpc_bv_pos_buf_size = 32768,
+
+            
+            disable_gmem = False,
+            gmem_size = 2 * 1024 * 1024,
+            shading_rate_matches_vk = True, #Экспериментально!!!
+            has_ray_intersection = False, 
+            enable_tp_ubwc_flag_hint = True,
         )],
         num_ccu = 4,
         num_slices = 2,
-        tile_align_w = 96,
+        tile_align_w = 64,
         tile_align_h = 32,
         tile_max_w = 16416,
         tile_max_h = 16384,
@@ -1455,8 +1538,37 @@ add_gpus([
         CHIP.A8XX,
         [a7xx_base, a7xx_gen3, a8xx_base, a8xx_gen2,
          GPUProps(
-            shading_rate_matches_vk = True,  # TODO confirm this
-            sysmem_vpc_bv_pos_buf_size = 24576,
+            # Sysmem кэши.
+             sysmem_vpc_attr_buf_size  = 131072,
+             sysmem_vpc_pos_buf_size = 65536,
+             sysmem_vpc_bv_pos_buf_size =  24576, # Значение стандартное, чтобы минимизировать артефакты 
+            # Sysmem глубина и цвет
+             sysmem_ccu_color_cache_fraction = CCUColorCacheFraction.FULL.value,
+             sysmem_per_ccu_color_cache_size = 128 * 1024,
+             sysmem_ccu_depth_cache_fraction = CCUColorCacheFraction.HALF.value,
+             sysmem_per_ccu_depth_cache_size = 128 * 1024,
+
+            # Gmem кэши (VPC)
+             gmem_vpc_attr_buf_size = 49152,
+             gmem_vpc_pos_buf_size = 24576,     
+             gmem_vpc_bv_pos_buf_size = 16384,  
+    
+    # Gmem глубина и цвет 
+              gmem_ccu_color_cache_fraction = CCUColorCacheFraction.HALF.value,
+              gmem_per_ccu_color_cache_size = 128 * 1024, 
+              gmem_ccu_depth_cache_fraction = CCUColorCacheFraction.HALF.value,
+              gmem_per_ccu_depth_cache_size = 128 * 1024,
+
+    # Оптимизации и фичи
+             has_ray_intersection = False, # По просьбе группы
+             has_sw_fuse = False,
+             has_coherent_ubwc_flag_caches = True,
+             has_fs_tex_prefetch = False,
+             has_salu_int_narrowing_quirk = True,
+             shading_rate_matches_vk = True, 
+             disable_gmem = False, 
+             gmem_size = 2 * 1024 * 1024,
+            enable_tp_ubwc_flag_hint = True,
          )],
         num_ccu = 4,
         num_slices = 2,
